elementary/clients/azure/client.py

```diff
@@ -0,0 +1,56 @@
+from os import path
+from typing import Optional, Tuple
+
+from azure.storage.blob import BlobServiceClient
+
+from elementary.config.config import Config
+from elementary.tracking.tracking_interface import Tracking
+from elementary.utils import bucket_path
+from elementary.utils.log import get_logger
+
+logger = get_logger(__name__)
+
+
+class AzureClient:
+    def __init__(self, config: Config, tracking: Optional[Tracking] = None):
+        self.config = config
+
+        self.tracking = tracking
+
+    @classmethod
+    def create_client(
+        cls, config: Config, tracking: Optional[Tracking] = None
+    ) -> Optional["AzureClient"]:
+        return cls(config, tracking=tracking) if config.has_blob else None
+
+    def send_report(
+        self, local_html_file_path: str, remote_bucket_file_path: Optional[str] = None
+    ) -> Tuple[bool, Optional[str]]:
+        report_filename = (
+            bucket_path.basename(remote_bucket_file_path)
+            if remote_bucket_file_path
+            else path.basename(local_html_file_path)
+        )
+        blob_service_client = BlobServiceClient.from_connection_string(
+            self.config.azure_connection_string
+        )
+        self.client = blob_service_client.get_blob_client(
```

Comment:
This is specific to a container and a blob, so it should not be set as the AzureClient self.client (each blob will have his own and it is not "general")